lib/_http_client.js

```diff
@@ -811,6 +817,8 @@ function tickOnSocket(req, socket) {
     parser.maxHeaderPairs = req.maxHeadersCount << 1;
   }
 
+  parser.joinDuplicateHeaders = req.joinDuplicateHeaders;
```

Comment:
why is this needed?

Reply:
To pass the `joinDuplicateHeader` option from client request to `parserOnHeadersComplete` . maybe there is some better way?